pkg/repository/v1/filters.go

```diff
@@ -0,0 +1,40 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/hatchet-dev/hatchet/pkg/repository/v1/sqlcv1"
+)
+
+type FilterRepository interface {
+	CreateFilter(ctx context.Context, params sqlcv1.CreateFilterParams) (*sqlcv1.V1Filter, error)
```

Comment:
We haven't been doing this recently but really should be -- I prefer that we create structs with custom validation params at the repository layer and call validate on these methods at this layer as well. The most fully-featured example is the methods which upsert a workflow version I think, which you can use as a reference.